src/plugins/visualizations/server/saved_objects/visualization_migrations.ts

```diff
@@ -981,6 +982,78 @@ const removeDefaultIndexPatternAndTimeFieldFromTSVBModel: SavedObjectMigrationFn
   return doc;
 };
 
+const addEmptyValueRuleForSavedObjectsWithLessAndGreaterThenZeroRules: SavedObjectMigrationFn<
+  any,
+  any
+> = (doc) => {
+  const visStateJSON = get(doc, 'attributes.visState');
+  let visState;
+
+  if (visStateJSON) {
+    try {
+      visState = JSON.parse(visStateJSON);
+    } catch (e) {
+      // Let it go, the data is invalid and we'll leave it as is
+    }
+
+    if (visState && visState.type === 'metrics') {
+      const params: any = get(visState, 'params') || {};
+      const {
+        bar_color_rules: barColorRules = [],
+        background_color_rules: backgroundColorRules = [],
+        gauge_color_rules: gaugeColorRules = [],
+      } = params;
+
+      const getRuleWithComparingToZero = (rules: any[] = []) => {
+        const compareWithEqualMethods = ['lte', 'gte'];
+        return last(
+          rules.filter(
+            (rule) => compareWithEqualMethods.includes(rule.operator) && rule.value === 0
+          )
+        );
+      };
+
+      const convertRuleToEmpty = (rule: any = {}) => ({
+        ...rule,
+        id: uuid.v4(),
+        operator: 'empty',
```

Comment:
nit: if I'm not wrong you created a constant for that in common folder. Let's use it

Reply:
I was thinking about it but didn't know if it is a good idea to import something to the migrations from other plugins.